File: go/src/pe010.go
package main

import (
	"fmt"
	"math"
)

func is_prime(test int, primes []int) bool {
	root := int(math.Sqrt(float64(test)))
	for i := 0; i < len(primes); i++ {
		if primes[i] > root {
			return true
		}
		if test%primes[i] == 0 {
			return false
		}
	}
	return false
}

func main() {
	last := 2000000
	primes := []int{2}

	var sum int
	sum = 2
	for i := 3; i < last; i = i + 2 {
		if is_prime(i, primes) {
			primes = append(primes, i)
			sum += i
		}
	}
	fmt.Printf("%d\n", sum)
}

File: typescript/pe010.ts
function is_prime(test:number, numprimes:number, primes:Array<number>): boolean {
  let root = Math.floor(Math.sqrt(test))
  for (let i=0; i<numprimes; i++)
  {
    if (primes[i] > root)
      return true
    if (!(test % primes[i]))
      return false
  }
  return false
}

function pe010(last:number): number {
  let primes:Array<number> = [2]
  let numprimes:number = 1
  let sum:number = 2
  
  for (let test = 3; test < last; test+=2)
  {
    if (is_prime(test, numprimes, primes))
    {
      primes[numprimes++] = test
      sum += test
    }
  }
  return sum
}

const last:number = 2000000

console.log(pe010(last))
